CATROID-864 Show keyboard for TextInputDialogs

mstoeg opened the pull request:
https://jira.catrob.at/browse/CATROID-864

- Fixed bug of Scratch2Catrobat search view, where the results of the old search persisted even if no more character was within the textfield.

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

ndrnour reviewed (approved):
well done. this will fix it for all alert dialogs. but not for custom dialogs that contain Editable textView like `RegistrationDialogFragment` and `LoginDialogFragment`, which should be fixed once we have a ticket to refactor them to kotlin.

wslany commented:
Very nice!

Two questions, just to make sure: 
1. Is this a behavior that can be tested? It would be great, so we don't forget it should work.
2. Also, I found that the dialog for the Scratch converter interface does not automatically display the keyboard for the search text field. It would be awesome to have the keyboard shown there too, since one would normally always need it and the screen anyway is half empty, since there are just four sample entries. However, if this should better be kept for another ticket, that would be OK from my side. So, only if it is straightforward with this ticket, if at all.

wslany commented:
If you change anything, please also update the license header date range to -2021.

wslany reviewed (changes requested):
As discussed on slack. Thank you!

mstoeg commented:
I implemented the requested changes regarding custom dialogs (like `RegistrationDialogFragment` and `LoginDialogFragment`) and for the `ScratchConverter` Activity. In the course of that I also changed the header date. 

As far as a testcase is concerned I could not really come up with a sensible way to test this. With a Espresso Test testing all the occurences would be extremely inefficient. Moreover, the tests would most likely be flaky because of timing varieties until the keyboard is displayed. Also a Unit Test makes no sense in this case as it would require to mock the whole building process of the dialogs which is not really feasible.

ndrnour commented:
it looks good, but some optimization can be done. let me know when you have time for a call in order to do it.

ndrnour commented:
regarding the testing, until now there is no way to check whether the keyboard is shown or hidden on Android.
Therefore no testing in this PR.

wslany commented:
Thanks!!
